src/main/java/com/amazon/jenkins/ec2fleet/EC2FleetCloud.java

```diff
@@ -665,35 +665,35 @@ public void run() {
     }
 
     /**
-     * Schedule Jenkins Node and EC2 instance to termination. Check first if target capacity more
-     * then <code>minSize</code> otherwise reject termination.
+     * Schedules Jenkins Node and EC2 instance for termination.
+     * If <code>force</code> is false, check if target capacity more than <code>minSize</code> otherwise reject termination.
+     * Else if <code>force</code> is true, schedule instance for termination even if it breaches <code>minSize</code>
      * <p>
-     * Real termination will happens in {@link EC2FleetCloud#update()} which periodically called by
-     * {@link CloudNanny}. So it could be some lag between decision that node should be terminated
+     * Real termination will happens in {@link EC2FleetCloud#update()} which is periodically called by
+     * {@link CloudNanny}. So there could be some lag between the decision to terminate the node
      * and actual termination, you can find max lag size in {@link CloudNanny#getRecurrencePeriod()}
      * <p>
      * This method doesn't do real termination to reduce load for Jenkins in case when multiple nodes should be
      * terminated in short time, without schedule process and batch termination, multiple calls should be raised
      * to AWS EC2 API which takes some time and block cloud class.
      *
      * @param instanceId node name or instance ID
-     * @return <code>true</code> if node scheduled to delete, otherwise <code>false</code>
+     * @param force terminate instance even if it breaches min size constraint
+     * @return <code>true</code> if node scheduled for termination, otherwise <code>false</code>
      */
-    public synchronized boolean scheduleToTerminate(final String instanceId) {
+    public synchronized boolean scheduleToTerminate(final String instanceId, final boolean force) {
```

Comment:
Using "force" that doesn't force schedule termination in all cases feels misleading. I think we should rename this to "ignoreMinSize" or something like that

Reply:
Doesn't it force schedules in all cases as long as the `stats` is set (which gets set in constructor now anyway)? Also longer term we may add more constraints (maybe few for spare-instances) apart from just minSize however we would still be terminating these instances regardless.

Reply:
true because we only have two cases right now :p (3 in label based fleet)

I suppose I was thinking about it in the reverse way for future cases, where maybe we would want to ignore the min size check but still consider other conditions. Easy to refactor at that point either way though so I'm fine with this